pkg/monitor/cluster/clusterwideproxystatus.go

```diff
@@ -0,0 +1,250 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v6"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apisubnet "github.com/Azure/ARO-RP/pkg/api/util/subnet"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+)
+
+const (
+	cwp             = "clusterWideProxy.status"
+	cwpErrorMessage = "NoProxy entries are incorrect"
+)
+
+var mandatory_no_proxies = "localhost,127.0.0.1,.svc,.cluster.local.,168.63.129.16,169.254.169.254"
+
+func contains(slice []string, item string) bool {
+	if strings.Contains(fmt.Sprint(slice), item) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (mon *Monitor) emitCWPStatus(ctx context.Context) error {
+	mon.hourlyRun = true
+	proxyConfig, err := mon.configcli.ConfigV1().Proxies().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		log.Fatalf("Error in getting the cluster wide proxy: %v", err)
+	}
+	if proxyConfig.Spec.HTTPProxy == "" && proxyConfig.Spec.HTTPSProxy == "" && proxyConfig.Spec.NoProxy == "" {
+		mon.emitGauge(cwp, 1, map[string]string{
+			"status":  strconv.FormatBool(false),
+			"Message": "CWP not enabled",
+		})
+	} else {
+		no_proxy_list := strings.Split(proxyConfig.Spec.NoProxy, ",")
```

Comment:
The `no_proxy_list` can be converted into a hashmap for a O(1) lookup each time we are searching for string in it. Something like a `map[string]struct{}`. This will get rid of calling `strings.Contains()` totally.